### ISSUE TITLE
bceloss provides too huge grad. Imo, it's bad.

### ISSUE THREAD

**Issue by YagaoDirac** [pytorch/pytorch#171327]
### 🐛 Describe the bug

```
bceloss_no_reduction = torch.nn.BCELoss(reduction="none")
input = torch.linspace(start=0., end=1., steps=5, requires_grad=True)
target = torch.zeros_like(input)
loss:torch.Tensor = bceloss_no_reduction(input, target)
print(loss)#[  0.0000,   0.2877,   0.6931,   1.3863, 100.0000]
loss.backward(torch.ones_like(loss))
print(input.grad)#[0.0000e+00, 1.3333e+00, 2.0000e+00, 4.0000e+00, 1.0000e+12]
```
The last element in grad is 1e12, it's too big. What learning rate should I use to get this to work?
I know in programming, or math, the implementation is correct. But in the perspective of training a model, most grad is in the range of (0,10), but sometimes it shoots up to a extremely big number.

I open this issue only to suggest you guys rethink about this detail. I know this is an open question. 
Simply close the issue. No action is needed immediately. 

### Versions

2.9
cuda 13

**Comment by YagaoDirac** [comment]
> I think you should try gradient clipping and also log the grad norm values during training.
> 
> Some articles and docs that you can refer to: https://neptune.ai/blog/understanding-gradient-clipping-and-how-it-can-fix-exploding-gradients-problem https://docs.pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_norm_.html

Yeah, I do a lot these work in training. I mean, this default setting is a bit not very friendly to beginners. Since torch is meant to be easy to use. This is not for me, but to make beginners' lives easier.
